### PR TITLE
fix(Text.Detail): forward `id` prop when `as` prop is not defined

### PR DESCRIPTION
**Pull request by ragafus**
Closes https://github.com/commercetools/ui-kit/issues/2867

#### Summary

Forwards `Text.Detail` `id` prop in all circumstances.

## Description

See full context in https://github.com/commercetools/ui-kit/issues/2867

### CODE REVIEW

**Review by misama-ct — APPROVED** [review]
Thank you 🙇

**Review by ddouglasz — APPROVED** [review]
Thanks! LGTM 💯

**Comment by ddouglasz** [comment]
No ux/ui changes. Doesn't require designer approval before merging.

**Pull request merged**
